src/pluginplay/printing/mermaid.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2024 NWChemEx-Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "module/module_class.hpp"
+#include "module_manager/module_manager_class.hpp"
+#include <pluginplay/pluginplay.hpp>
+#include <sstream>
+
+void print_submods(const std::string module_key,
+                   const pluginplay::ModuleManager& mm, std::stringstream& ss,
+                   int level = 0) {
+    const auto& mm_module = mm.at(module_key); // Results in a PluginPlay Module
+    const auto& submods = mm_module.submods(); // Results in list of Submodules
+    char letter         = 65 + level;
+
+    // Key is the ID/Tag, Value is the reference to the Module
+    for(const auto& [key, value] : submods) {
+        ss << "\t" << letter + level << level + 1 << "[" << key
+           << "]\n"; // Second indent: ----
+        if(value.has_module() == false) {
+            ss << "\t" << letter + level << level + 2
+               << "[Submod Name: No Submodule associated with Key]\n";
+            continue;
```

Comment:
You don't need the `continue` statement here (the rest of the `for` loop is in the `else` branch which won't be run.

```suggestion
```